conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/processing/BakersProcessing.scala

```diff
@@ -0,0 +1,246 @@
+import scala.util.{Failure, Success, Try}
+import akka.Done
+import akka.stream.scaladsl.Source
+import tech.cryptonomic.conseil.indexer.config.{BakingAndEndorsingRights, BatchFetchConfiguration}
+import tech.cryptonomic.conseil.common.tezos.Tables
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.{
+  Block,
+  BlockReference,
+  BlockTagged,
+  Delegate,
+  InvalidPositiveDecimal,
+  PositiveBigNumber,
+  PositiveDecimal,
+  PublicKeyHash,
+  Voting
+}
+import tech.cryptonomic.conseil.indexer.tezos.{TezosNodeOperator, TezosDatabaseOperations => TezosDb}
+import tech.cryptonomic.conseil.indexer.tezos.TezosNodeOperator.LazyPages
+import tech.cryptonomic.conseil.indexer.tezos.TezosErrors.BakersProcessingFailed
+import com.typesafe.scalalogging.LazyLogging
+import slick.jdbc.PostgresProfile.api._
+import akka.stream.ActorMaterializer
+
+/** Takes care of fetching and processing any bakers' related data.
+  *
+  * @param nodeOperator
+  * @param db
+  * @param batchingConf
+  */
+class BakersProcessing(
+    nodeOperator: TezosNodeOperator,
+    db: Database,
+    batchingConf: BatchFetchConfiguration,
+    rightsConf: BakingAndEndorsingRights
+)(implicit mat: ActorMaterializer)
+    extends LazyLogging {
+
+  /* Fetches the data from the chain node and stores bakers into the data store.
+   * @param ids a pre-filtered map of delegate hashes with latest block referring to them
+   */
+  private def process(ids: Map[PublicKeyHash, BlockReference], onlyProcessLatest: Boolean = false)(
+      implicit ec: ExecutionContext
+  ): Future[Done] = {
+    import cats.Monoid
+    import cats.instances.future._
+    import cats.instances.int._
+    import cats.instances.option._
+    import cats.syntax.flatMap._
+    import cats.syntax.monoid._
+
+    def logWriteFailure: PartialFunction[Try[_], Unit] = {
+      case Failure(e) =>
+        logger.error(s"Could not write bakers to the database", e)
+    }
+
+    def logOutcome: PartialFunction[Try[(Option[Int], Option[Int])], Unit] = {
+      case Success((rows, historyRows)) =>
+        logger.info("{} bakers were touched on the database.", rows.fold("The")(String.valueOf))
+        logger.info("{} baker history rows were added to the database.", historyRows.fold("The")(String.valueOf))
+    }
+
+    def processBakersPage(
+        taggedBakers: Seq[BlockTagged[Map[PublicKeyHash, Delegate]]],
+        rolls: List[Voting.BakerRolls]
+    ): Future[(Option[Int], Option[Int])] = {
+
+      val enrichedBakers = taggedBakers
+        .map(
+          blockTagged =>
+            blockTagged.copy(content = blockTagged.content.map {
+              case (key, baker) =>
+                val rollsSum = rolls.filter(_.pkh.value == key.value).map(_.rolls).sum
+                (key, baker.updateRolls(rollsSum))
+            })
+        )
+
+      db.run(TezosDb.writeBakersAndCopyContracts(enrichedBakers.toList))
+        .andThen(logWriteFailure)
+    }
+
+    def cleanup[T] = (_: T) => {
```

Comment:
This signature looks weird, do we need to have this `(_: T)` parameter?

Reply:
Not necessarily? I'll check it out, thanks